default-fields/0000-default-fields.md

```diff
@@ -28,9 +233,29 @@ Why should we *not* do this?
 # Alternatives
 [alternatives]: #alternatives
 
-What other designs have been considered? What is the impact of not doing this?
+Just stick with builders and constructor functions.
 
 # Unresolved questions
 [unresolved]: #unresolved-questions
 
-What parts of the design are still TBD?
+## `const`
+
+Is `const` the right feature to base off?
+It means field values are just as constrained as `const` is now,
+but also prevents struct definitions from becomes bloated by complex logic that
+maybe belongs in a builder instead.
+
+## When all fields have defaults
+
+An interesting case is where all fields are given default values:
+
+```rust
+struct Foo {
+    a: &'static str = "",
+    b = true
+    c: i32 = 42
+}
+```
+
+Should this type automatically derive `Default`?
+Should we be able to construct this type as if it's a ZST, like `let foo = Foo;`?
```

Comment:
This is a unit struct, you can also use `Foo {}` for an empty struct, and the braces must match - you can't use one declaration with the other construction. That suggests you can't use `Foo`, but need `Foo{}` (also note that the two forms do have slightly different rules for naming).